shadowtools/src/shadowtools/shadow_exec.py

```diff
@@ -111,11 +113,20 @@ def _main(
     config_path.write_text(yaml.safe_dump(config))
 
     data_dir = tmpdir.joinpath("shadow.data")
-    shadow_args = [
-        f"--data-directory={data_dir}",
-        "--progress=false",
-        str(config_path),
-    ]
+    if any((re.match(r"^--data-directory[=$]|^-d", s) for s in shadow_args)):
```

Comment:
I'm not sure that this regex is working as intended. For example the following returns `None`.

```python3
re.match(r"^--data-directory[=$]|^-d", "--data-directory")
```